constraint_utils/constraint_basics.py

```diff
@@ -378,6 +378,40 @@ def keep_criteria(ht):
 
 def build_models(coverage_ht: hl.Table, trimers: bool = False, weighted: bool = False, half_cutoff = False,
                  ) -> Tuple[Tuple[float, float], Dict[str, Tuple[float, float]]]:
+    """
+    Build coverage model and plateau models.
+    
+    This function first removes possible variants where there was no coverage, a low-quality variant was observed,
+    or a variant above 0.1% frequency was observed, and then calibrate from the mutation rate to proportion observed
+    to build plateau models for sites above `HIGH_COVERAGE_CUTOFF`. 
+    
+    For sites below `HIGH_COVERAGE_CUTOFF`, this function performs a base-level resolution. It defines a metric
+    that is derived by dividing the number of observed synonymous variants with the total number of possible synonymous
+    variants times the mutation rate summed across all substitutions, contexts, and methylation level. The function computes
+    this metric for high coverage sites as a global scaling factor, and divide this metric at low coverage sites by this 
+    scaling factor to create an observed:expected ratio. The function finally builds a coverage model of log10(coverage) to
+    this scaled ratio as a correction factor for low coverage sites.
+    
+    As input, `coverage_ht` is the output of `get_proportion_observed_by_coverage`.
+    
+    .. note::
+        The following fields should be present in `coverage_ht`:
+            - context - a codon
+            - ref - the middle base on the reference
+            - alt - the alternate base
+            - methylation_level
+            - exome_coverage
+            - variant_count - the number of observed variants
+            - downsampling_counts_{pop} (pop defaults to `POPS`) - array of observed variant counts per population after downsampling
+            - mu_snp - mutation rate
+            - possible_variants
```

Comment:
I wonder if it would be better to move this to right after the one line summary and before the full description of what the function does.

Also missing `possible_variants`

```suggestion
    .. note::
        This function expects that the input `coverage_ht` is the output of `get_proportion_observed_by_coverage`, and therefore the following fields should be present in `coverage_ht`:
            - context - trinucleotide genomic context
            - ref - the middle base of `context`
            - alt - the alternate base
            - methylation_level - methylation level
            - exome_coverage - median exome coverage at integer values between 1-100
            - variant_count - the number of observed variants in the dataset for each substitution (`alt`), `context`, `methylation_level`, and median exome coverage (`exome_coverage`)
            - downsampling_counts_{pop} (pop defaults to `POPS`) - array of observed variant counts per population after downsampling
            - mu_snp - mutation rate
            - possible_variants
```

Reply:
I prefer to leave the notes at the end, so that the reader could have an overview of what the function does before going into details like what annotation the input table will have. Also it will keep the same pattern as `prepare_ht()` did.